frontend/src/component/common/Combo/Combo.tsx

```diff
@@ -0,0 +1,150 @@
+    )
+        return <Autocomplete {...props} />;
+
+    return (
+        <Autocomplete
+            {...props}
+            disableListWrap
+            ListboxComponent={ListboxComponent}
+        />
+    );
+};
+
+const ListboxComponent = forwardRef<
+    HTMLDivElement,
+    HTMLAttributes<HTMLElement>
+>(function ListboxComponent(props, ref) {
+    const virtuosoRef = useRef<GroupedVirtuosoHandle>(null);
+    const [currentItemIndex, setCurrentItemIndex] = useState(-1);
+
+    const elements = (props.children || []) as JSX.Element[];
+
+    const groups = elements.map(({ props }) => {
+        const [header, list] = props.children;
+        const items = list.props.children;
+        return {
+            total: items.length,
+            header,
+            items,
+        };
+    });
+
+    const groupCounts = groups.map(({ total }) => total);
+
+    const keyDownCallback = useCallback(
+        e => {
+            let nextIndex: number | null = null;
+
+            if (e.code === 'ArrowUp') {
+                nextIndex = Math.max(0, currentItemIndex - 1);
+            } else if (e.code === 'ArrowDown') {
+                nextIndex = Math.min(
+                    groupCounts.reduce((a, b) => a + b, 0),
+                    currentItemIndex + 1
+                );
+            }
+
+            if (nextIndex !== null) {
+                setCurrentItemIndex(nextIndex);
+                virtuosoRef.current?.scrollIntoView({
+                    index: nextIndex,
+                    behavior: 'auto',
+                });
+                e.preventDefault();
+            }
+        },
+        [currentItemIndex, virtuosoRef, setCurrentItemIndex]
+    );
+
+    const selectItem = useCallback(
+        e => {
+            const li = e.target.closest('li');
+            if (li) setCurrentItemIndex(+li.dataset.optionIndex);
+        },
+        [currentItemIndex, virtuosoRef, setCurrentItemIndex]
+    );
+
+    const scrollUp = useCallback(() => {
+        virtuosoRef.current?.scrollToIndex({
+            index: 0,
+            behavior: 'auto',
+        });
+    }, [virtuosoRef]);
+
+    const scrollerRef = useCallback(
+        element => {
+            if (element) {
+                window.addEventListener('keydown', keyDownCallback);
+                window.addEventListener('click', selectItem);
+                window.addEventListener('keypress', scrollUp);
+            } else {
+                window.removeEventListener('keydown', keyDownCallback);
+                window.removeEventListener('click', selectItem);
+                window.removeEventListener('keypress', scrollUp);
+            }
+        },
+        [keyDownCallback]
+    );
+
+    return (
+        <StyledDiv ref={ref} {...props}>
+            <GroupedVirtuoso
+                ref={virtuosoRef}
+                scrollerRef={scrollerRef}
+                increaseViewportBy={{ top: 100, bottom: 100 }}
```

Comment:
This improves keyboard navigation a bit by having some more items pre-rendered out of view.